go/api/command_options.go

```diff
@@ -18,10 +18,11 @@ type SetOptions struct {
 	// Equivalent to GET in the valkey API.
 	ReturnOldValue bool
 	// If not set, no expiry time will be set for the value.
+	// Supported ExpiryTypes ("EX", "PX", "EXAT", "PXAT", "KEEPTTL")
```

Comment:
A user can use `PERSIST` there and get an error. Why not to restrict that in compile time?

You probably need to reorganize these options, `Expiry` and `ExpiryType` to ensure that expiry value should be provided with "EX", "PX", "EXAT", and "PXAT" only.

ts-style unions or rust-style enums perfectly fit in such case.

Reply:
Returning an error in the response if the user has provided wrong expiry type.